src/lookup/query.test.ts

```diff
@@ -180,6 +171,30 @@ describe("query", () => {
       },
     );
 
+    it("throws if the data is too large to process as JSON", async () => {
+      const mockVp = JSON.stringify(mockDefaultPresentation());
+      const mockedFetch = jest
+        .fn<typeof fetch>()
+        .mockResolvedValueOnce(createResponse(mockVp));
+
+      const defaultMaxJsonSize = getMaxJsonSize();
+      setMaxJsonSize(10);
+      await expect(
+        query(
+          "https://some.endpoint/query",
+          { query: [mockRequest] },
+          {
+            fetch: mockedFetch,
+            returnLegacyJsonld: false,
+          },
+        ),
+      ).rejects.toThrow(
+        `The holder [https://some.endpoint/query] did not return a valid JSON response: parsing failed with error ` +
+          `Error: The response body is not safe to parse as JSON. Max size=[10], actual=[${mockVp.length}]`,
+      );
+      setMaxJsonSize(defaultMaxJsonSize);
```

Comment:
If you want to avoid requiring the reset here, I think you can put this test in a separate `describe` block, which will have it run in a separate environment where it will have no side effects on other tests.

Reply:
Didn't work as expected so leaving this